mqtt/mqtt-edgehub/src/command.rs

```diff
@@ -109,33 +103,37 @@ impl CommandHandler {
                 .map_err(CommandHandlerError::SubscribeFailure)?;
         }
 
-        let mut subacks: HashSet<String> = HashSet::from_iter(topics.to_vec());
+        let mut subacks: Vec<String> = topics.to_vec();
 
-        while let Some(event_or_err) = self
+        while let Some(event) = self
             .client
             .try_next()
             .await
             .map_err(CommandHandlerError::PollClientFailure)?
         {
-            if let Event::SubscriptionUpdates(subscriptions) = event_or_err {
+            if let Event::SubscriptionUpdates(subscriptions) = event {
                 for subscription in subscriptions {
                     if let SubscriptionUpdateEvent::Subscribe(sub) = subscription {
-                        subacks.remove(&sub.topic_filter);
+                        subacks.retain(|topic| *topic != sub.topic_filter);
```

Comment:
There might be a better way to do this. It will keep all topics that don't match the incoming suback. So, it will remove all instances of the topic from the vector.

Reply:
Oops, yeah, I missed that. Ordering is not important here, so we can use [swap_remove](https://doc.rust-lang.org/std/vec/struct.Vec.html#method.swap_remove)